packages/terminal/src/widget.ts

```diff
@@ -176,8 +177,10 @@ export class Terminal extends Widget implements ITerminal.ITerminal {
         });
       }
     }
-    this._term.dispose();
-    super.dispose();
+    if (this.getOption('disposeOnExit')) {
+      this._term.dispose();
+      super.dispose();
+    }
```

Comment:
For instance, when we exit a terminal with `disposeOnExit=false`, and restart JupyterLab, the workspace is restored with a terminal widget that has no terminal in the back-end:

![image](https://user-images.githubusercontent.com/4711805/145235636-ea302e29-8490-4475-ae93-2a46e9280e6d.png)

We have this message in the back-end:
```
[W 2021-12-08 16:26:28.658 ServerApp] 404 GET /terminals/websocket/1 (127.0.0.1) 30.48ms referer=None
```

Maybe the behavior should be to not reopen this widget?

Reply:
Maybe the `disposeOnExit` check should be done on `session.disposed` here:

https://github.com/jupyterlab/jupyterlab/blob/5590af614f4585138cbd19cffe98905faf35bf24/packages/terminal/src/widget.ts#L74

To avoid preventing the calls to the parent `dispose`.